### PR TITLE
NEW: Improve job list in admin/queuedjobs

### PR DESCRIPTION
**Pull request by sminnee**
- Translate the job type from a number to a string
 - Add “finished”, which can be useful for seeing job duration

### CODE REVIEW

File: src/DataObjects/QueuedJobDescriptor.php [hunk 1]
```diff
@@ -284,6 +285,21 @@ public function getTitle()
...
+     * Return a string representation of the numeric JobType
+     * @return string
+     */
+    public function getJobTypeString()
+    {
+        $map = [
+            QueuedJob::IMMEDIATE => 'Immediate',
+            QueuedJob::QUEUED => 'Queued',
+            QueuedJob::LARGE => 'Large',
+        ];
```
Comment (robbieaverill) [inline src/DataObjects/QueuedJobDescriptor.php]: Do you think we should make these translatable as well?

Comment (sminnee) [reply src/DataObjects/QueuedJobDescriptor.php]: Yeah, makes sense.

**Comment by sminnee** [comment]
The "finished" column is not hugely useful unless max_finished_jobs_age is reasonably long. By default it's 300 (5 minutes). On the project I have most to do with queuedjobs on, I set this to 7200 (2 hours) which gives you a better view of recent jobs in the admin.

Should we increase the default value of max_finished_jobs_age too?

**Comment by robbieaverill** [comment]
I think it'd totally depend on what kind of project you have, which would probably be how many jobs you run regularly. I think 5 mins is probably useless for most people so yeah I'd be happy to increase it to a couple of hours.

File: src/DataObjects/QueuedJobDescriptor.php (hunk 1 above)
Comment (sminnee) [reply src/DataObjects/QueuedJobDescriptor.php]: Done.

File: src/DataObjects/QueuedJobDescriptor.php [hunk 2]
```diff
@@ -284,6 +285,30 @@ public function getTitle()
...
+    /**
+     * Return a map of numeric JobType values to localisable string representations.
+     * @return array
+     */
+    public function getJobTypeValues()
+    {
+        return [
+            QueuedJob::IMMEDIATE => _t(QueuedJob::class . '.TYPE_IMMEDIATE', 'Immediate'),
+            QueuedJob::QUEUED => _t(QueuedJob::class . '.TYPE_QUEUED', 'Queued'),
+            QueuedJob::LARGE => _t(QueuedJob::class . '.TYPE_LARGE', 'Large'),
```
Comment (robbieaverill) [inline src/DataObjects/QueuedJobDescriptor.php]: Unfortunately `Some::class` doesn't work in the text collector. Will need to be `_t('Symbiote\\QueuedJobs\\Services\\QueuedJob',`

Comment (sminnee) [reply src/DataObjects/QueuedJobDescriptor.php]: Doh, you're right https://github.com/silverstripe/silverstripe-framework/issues/7647

I might just use `__class__`; these values will be attached to QueuedJobDescriptor but that's not a showstopper?

Comment (robbieaverill) [reply src/DataObjects/QueuedJobDescriptor.php]: Yep sounds good

Comment (sminnee) [reply src/DataObjects/QueuedJobDescriptor.php]: Done; squashed in.

**Pull request merged**